### ISSUE TITLE
Bad toolbar positioning

### ISSUE THREAD

**Issue by fredck** [ckeditor/ckeditor5-ui#66]
There are two issues with the toolbar floating and sticking positioning. Both have been identified at the [online demo](https://ckeditor5.github.io/).

## Issue 1: Editor needs activation

 1. Load the demo without clicking the editor.
 2.  Scroll down, so the editor moves out of the view. Issue: the toolbar doesn't get stuck to the top border of the view.
 3. With the editor partially out of view, click inside of it. The toolbar will suddenly appear stuck now and the scrolling will work as expected.

## Issue 2: Toolbar doesn't stay with the editor

 1. Click inside the editor.
 2. Scroll down until the editor disappears. Issue: the toolbar will not stop at the minimum editor height and instead will remain stuck, even when the editor gets fully hidden.

**Comment by oleq** [comment]
Issue 1, which is

> the toolbar doesn't get stuck to the top border of the view

is not actually an issue but a conscious decision. No interaction with the editor (no focus in editable) means the UI remains 100% passive (static). Only focused editors can have an active UI which follows the user, etc. Imagine what would happen if there were 5 editors in the web page, none of them focused like in the TC and the user would start scrolling the web page. If the UI followed your algorithm, soon you'd have 5 toolbars on the top of the webpage, one next (upon?) another, like this

![Toolbar paradise](https://cloud.githubusercontent.com/assets/1099479/18092939/7b379830-6ece-11e6-84f5-22544052280a.png)


As for the Issue 2 – it is known to me. It was ignored in the initial implementation for the sake of faster development and unblocking other features.

So I'm for fixing number 2 in this issue.

**Comment by fredck** [comment]
Thanks for the opinion @oleq, but your point about Issue 1 is wrong exactly because of Issue 2, which will hide the toolbars of editors that are not in the viewport anymore. Therefore having Issue 1 and 2 fixed at the same time will bring an adequate user experience.

**Comment by oleq** [comment]
That's a good point, @fredck, but still I think Issue 1 is invalid because it brings nothing more to the user but an "everything moves" feeling. What are the real pros of the toolbar following the viewport if the user is not interested in that particular editor instance? It only slows down the webpage IMO and brings no richer UX than static toolbar.

**Comment by Reinmar** [comment]
If we change behaviour 1, then this induces changing the behaviour 2. But the opposite is not true – if we fix 2, we don't have to fix 1. And I agree with Olek that only 2 is a real issue.

That makes behaviour 1 optional, but not required. I agree that in some scenarios it may be what you want – e.g. you have one, huge editor on a page and wants the toolbar to be always visible, regardless whether the focus is in the editor UI or your app's UI. We only need to decide what should be the default of this option (cause it sounds like a config option).

**Comment by Reinmar** [comment]
tl;dr: We shouldn't be talking about fixing behaviour 1, but making it optional. There are cases when inactive toolbars shouldn't follow you.

**Comment by fredck** [comment]
Confirmed that Issue 1 is not a requirement, but IMHO if we'll ever go for it, let's not introduce configurations for the sake of having them. If we have Issue 1 fixed there will be no case where it will be wrong.

For sure there will be cases of people clicking in the contents and the toolbar suddenly covering the selection (maybe even unwanted toolbar clicks when attempting double clicks). That, together with the surprise, may be the only drawbacks of the current implementation.
